command/genesis/polybft_params.go

```diff
@@ -45,28 +43,78 @@ const (
 )
 
 func (p *genesisParams) generatePolyBFTConfig() (*chain.Chain, error) {
-	validatorsInfo, err := ReadValidatorsByRegexp(path.Dir(p.genesisPath), p.polyBftValidatorPrefixPath)
+	// set initial validator set
+	genesisValidators, err := p.getGenesisValidators()
 	if err != nil {
 		return nil, err
 	}
 
+	// deploy genesis contracts
 	allocs, err := p.deployContracts()
 	if err != nil {
 		return nil, err
 	}
 
-	// use 1st account as governance address
-	governanceAccount := validatorsInfo[0].Account
+	// premine accounts with some tokens
+	var (
+		validatorPreminesMap map[types.Address]int
+		premineInfos         []*premineInfo
+	)
+
+	if p.premineValidators != "" {
+		validatorPreminesMap = make(map[types.Address]int, len(genesisValidators))
+
+		for i, vi := range genesisValidators {
+			premineInfo, err := parsePremineInfo(fmt.Sprintf("%s:%s", vi.Address, p.premineValidators))
+			if err != nil {
+				return nil, err
+			}
+
+			premineInfos = append(premineInfos, premineInfo)
+			validatorPreminesMap[premineInfo.address] = i
+		}
+	}
+
+	if len(p.premine) > 0 {
```

Comment:
I believe this if check is not necessary, you can just use for loop.

Reply:
Fixed https://github.com/0xPolygon/polygon-edge/pull/985/commits/414c6bde952d5b6768307fe53b08bb81c505a643